contracts/src/token/erc1155/extensions/uri_storage.rs

```diff
@@ -0,0 +1,85 @@
+//! ERC-1155 token with storage based token URI management.
+//!
+//! Inspired by the [`crate::token::erc721::extensions::Erc721UriStorage`]
+use alloc::string::String;
+
+use alloy_primitives::U256;
+use alloy_sol_types::sol;
+use stylus_proc::{public, sol_storage};
+use stylus_sdk::evm;
+
+sol! {
+
+    /// Emitted when the URI for token type `token_id` changes to `value`, if it is a non-programmatic URI.
+    ///
+    /// If an [`URI`] event was emitted for `token_id`, the standard
+    /// https://eips.ethereum.org/EIPS/eip-1155#metadata-extensions[guarantees] that `value` will equal the value
+    /// returned by [`Self::uri`].
+    #[allow(missing_docs)]
+    event URI(string value, uint256 indexed token_id);
+}
+
+sol_storage! {
+    /// Uri Storage.
+    pub struct Erc1155UriStorage {
```

Comment:
I see that you're not using `Erc1155UriStorage` as well as `Erc1155MetadataURI` in the examples and there is no e2e tests for it.
I would vote to not have any of those metadata extensions for `Erc1155` as a part of this pr.
I think we can add them later since I see there're problems around their design. And we'll being able to ship actual `Erc1155` standard faster. Wdt?
cc @ggonzalez94 @bidzyyys

Reply:
Agreed on keeping it simpler and shipping sooner since we have community demand for this